### ISSUE TITLE
Can not customize changelogPreset

### ISSUE THREAD

**Issue by Mister-Hope** [lerna-lite/lerna-lite#852]
### Describe the bug

It says here: https://github.com/lerna-lite/lerna-lite/tree/main/packages/version#--changelog-preset

While I have 

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/1b6b0dbc771a0e01c686cf94a1a120e6b011b517/lerna.json#L13-L77

From the logs:

```
lerna-lite info getChangelogConfig Auto-prefixing conventional-changelog preset "conventionalcommits"
lerna-lite info getChangelogConfig Successfully resolved preset "conventional-changelog-conventionalcommits"
```

It loads `conventional-changelog-conventionalcommits` successfuly, but I don't think any option is loaded, non of the `header` and `releaseCommitMessageFormat` configuration is working.

### Expectation

changelog header and commit msg should be customized.

### Reproduction

See https://github.com/vuepress-theme-hope/vuepress-theme-hope

Config are at https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/main/lerna.json#L13-L77

Command is `pnpm lerna version`

The result of the commit is at: https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/1b6b0dbc771a0e01c686cf94a1a120e6b011b517

The full log here:

```
> vuepress-theme-hope-root@2.0.0-rc.38 release:version /home/mister-hope/projects/vuepress/vuepress-theme-hope
> lerna version

lerna-lite notice cli v3.3.3
lerna-lite info current project version 2.0.0-rc.38
lerna-lite info Looking for changed packages since v2.0.0-rc.38
lerna-lite info ignoring diff in paths matching **/__tests__/** **/*.md
lerna-lite info getChangelogConfig Auto-prefixing conventional-changelog preset "conventionalcommits"
lerna-lite info getChangelogConfig Successfully resolved preset "conventional-changelog-conventionalcommits"

Changes (7 packages):
 - vuepress-plugin-components: 2.0.0-rc.38 => 2.0.0-rc.39
 - create-vuepress-theme-hope: 2.0.0-rc.38 => 2.0.0-rc.39
 - vuepress-plugin-lightgallery: 2.0.0-rc.38 => 2.0.0-rc.39
 - vuepress-plugin-md-enhance: 2.0.0-rc.38 => 2.0.0-rc.39
 - vuepress-plugin-search-pro: 2.0.0-rc.38 => 2.0.0-rc.39
 - vuepress-shared: 2.0.0-rc.38 => 2.0.0-rc.39
 - vuepress-theme-hope: 2.0.0-rc.38 => 2.0.0-rc.39

? Are you sure you want to create these versions? Yes
lerna-lite info git Pushing tags...
lerna-lite info execute Skipping releases
lerna-lite success version finished
```

### Environment Info

node v20.12.2
@lerna-lite/cli 3.3.3
@lerna-lite/publish 3.3.3

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/lerna-lite/lerna-lite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/lerna-lite/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/lerna-lite/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/lerna-lite/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
I never used that option, it came from the original Lerna. I'm trying to keep the project up to date by using Renovate but all the conventional-changelog dependencies were updated just a few days ago as major versions and none of them were upgraded yet as you can see in the Renovate [Dashboard](https://github.com/lerna-lite/lerna-lite/issues/434). I would suggest you downgrade your `conventional-changelog-conventionalcommits` from 8.0 to probably 7.x

I've answered similar question in this Stack Overflow [answer](https://stackoverflow.com/a/77914359/1212166), but like I said, I have never used this feature personally (though I do use `conventional-changelog-conventionalcommits` for its breaking change detection with something like `feat!` and I actually used it last week but I'm on version 7.x).

**Comment by Mister-Hope** [comment]
For me 7.x is also not working, I will try to do that and provide `--dry-run` log later

**Comment by ghiscoding** [comment]
I just tried it in dry-run mode and the types override works properly but considering that you're trying to use `releaseCommitMessageFormat` which seem to exists in [conventional-changelog-config-spec](https://github.com/conventional-changelog/conventional-changelog-config-spec/blob/master/versions/2.2.0/README.md#releasecommitmessageformat-string), however I wouldn't be surprised that this property cannot be overridden because of `command.version.message`. However, it's just a guess because like I said that was implemented by the original author, but since the types override do work, that's the only guess I could come up with

![image](https://github.com/lerna-lite/lerna-lite/assets/643976/44353936-f51e-4f56-97e3-945373c784ac)

**Comment by Mister-Hope** [comment]
> I just tried it in dry-run mode and the types override works properly (with emojis) but considering that you're trying to use `releaseCommitMessageFormat` which seem to exists in [conventional-changelog-config-spec](https://github.com/conventional-changelog/conventional-changelog-config-spec/blob/master/versions/2.2.0/README.md#releasecommitmessageformat-string), however I wouldn't be surprised that this property cannot be overridden because of `command.version.message` (have you tried to not set the prop?). However, it's just a guess because like I said that was implemented by the original author, but since the types override do work, that's the only guess I could come up with.

Thanks for the feedback, actually I am not focusing types with v7, but I think both commitmsg and header are not working.

**Comment by Mister-Hope** [comment]
update:

I have downgrade to v7 and update config like you suggest in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/e192e704c004ccec71a97c21e1e71e552e0d7f28. I remove the `command.version.message`, and I update `feat` and `feature` header with test.

Dry run log:

```
mister-hope@MrHope-Lenovo:~/projects/vuepress/vuepress-theme-hope$ pnpm release:version --dry-run

> vuepress-theme-hope-root@2.0.0-rc.38 release:version /home/mister-hope/projects/vuepress/vuepress-theme-hope
> lerna version "--dry-run"

lerna-lite notice cli v3.3.3
lerna-lite info [dry-run] > git rev-parse
lerna-lite info current project version 2.0.0-rc.41
lerna-lite info [dry-run] > git rev-list --count --all --max-count=1
lerna-lite info [dry-run] > git show-ref --verify refs/remotes/origin/dry-run
lerna-lite info [dry-run] > git remote update
lerna-lite info [dry-run] > git rev-list --left-right --count origin/dry-run...dry-run
lerna-lite info Looking for changed packages since v2.0.0-rc.40
lerna-lite info ignoring diff in paths matching **/__tests__/** **/*.md
lerna-lite info [dry-run] > git describe --always --long --dirty --first-parent
lerna-lite info getChangelogConfig Auto-prefixing conventional-changelog preset "conventionalcommits"
lerna-lite info getChangelogConfig Successfully resolved preset "conventional-changelog-conventionalcommits"

Changes (1 packages):
 - create-vuepress-theme-hope: 2.0.0-rc.41 => 2.0.0-rc.42

? [dry-run] Are you sure you want to create these versions? Yes
lerna-lite info [dry-run] > git add -- packages/create/package.json packages/create/CHANGELOG.md pnpm-lock.yaml CHANGELOG.md lerna.json
lerna-lite info [dry-run] > git commit -m v2.0.0-rc.42
lerna-lite info [dry-run] > git tag v2.0.0-rc.42 -m v2.0.0-rc.42
lerna-lite info git Pushing tags...
lerna-lite info [dry-run] > git push --follow-tags --no-verify --atomic origin dry-run
lerna-lite info execute Skipping releases
lerna-lite success version finished
```

I commited the result in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/523b1931a003966a729e3e2236a19554e9f37cd1

The types are working! But I don't think the `header` or the `releaseCommitMessageFormat` is working.

**Comment by Mister-Hope** [comment]
Also I confirm the `commit.version.message` is not working even if I delete `releaseCommitMessageFormat`, so it just seems only `types` are working for me, but I need to adjust the header, and format the commit msg

![image](https://github.com/lerna-lite/lerna-lite/assets/33315834/942e7af4-8d25-44b9-bb67-3b1d0e7a231c)

**Comment by ghiscoding** [comment]
I don't think the `releaseCommitMessageFormat` was meant to ever be supported, why can't you just use `command.version.message`? This message is provided to conventional changelog in every case and conventional-changelog is probably not using `releaseCommitMessageFormat` when `message` is also provided, I also don't want to change all the Lerna code related to conventional changelog... but really `{{currentTag}}` seems to be exactly the same as `%v` in `message`, so why not simply use that?

**Comment by Mister-Hope** [comment]
what about the header?

**Comment by ghiscoding** [comment]
Most probably the same, have you tried [`version --changelog-header-message`](https://github.com/lerna-lite/lerna-lite/tree/main/packages/version#--changelog-header-message-msg)? That's what I use in Lerna-Lite itself

**Comment by Mister-Hope** [comment]
yes, `command.version.changelogHeaderMessage` is working, thanks.

**Comment by Mister-Hope** [comment]
The header options is not working as expected, see https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/d86a705901f306a75513b21ef7894e33ce4ac407

![image](https://github.com/lerna-lite/lerna-lite/assets/33315834/e5fe4240-1818-4fa6-8dd0-5149da7ff69d)

**Comment by ghiscoding** [comment]
Yes that is the expected behavior since the common text of conventional commit "All behavior.." is a static text. The behavior of `changelogHeaderMessage` is really to prepend that text before the static conventional commit message, it is not meant to override the entire text but really to prepend it. I don't understand why you're trying to override a very similar message (you seem to only have 1 less line than original static message). 

It's probably a good idea to support the `header` message eventually, but I don't have time at the moment to look at this further

**Comment by Mister-Hope** [comment]
I want to insert the magic comment after the description,  That means I can import the final region I want  Of the change log,  So that I can show it in my documentation site.

So if the change log is actually a web page,  Then I don't think the description is needed or necessary,  Also I would want another heading for that.  Currently the magic comment can be inserted but only before the description.

**Comment by ghiscoding** [comment]
Like I said, it's the expected behavior, you can see the result of this behavior in Lerna-Lite [changelog](https://github.com/lerna-lite/lerna-lite/blob/main/CHANGELOG.md)

**Comment by ghiscoding** [comment]
This should be fixed in [v3.4.0](https://github.com/lerna-lite/lerna-lite/releases/tag/v3.4.0) but please note that as I mentioned earlier, I only added support for `changelogPreset.header` but did not support `releaseCommitMessageFormat` (though I now throw with an error mentioning to simply use `version.message` since the result is the same). I also saw from your config that you also wanted `changelogPreset.header` to work with an empty string, that will also work as shown below (I tried with `--dry-run` and below is the git diff of the change, notice the header is completely removed because it is set to an empty string)

![image](https://github.com/lerna-lite/lerna-lite/assets/643976/cf0720fa-0aa3-4898-9a03-6b4b187c86d9)

**Comment by ghiscoding** [comment]
@Mister-Hope just a side note, I see that you released a new version of your project couple days ago and you might be wondering why it didn't work, but the fact is that you did not update to Lerna-Lite 3.4.0 yet so it won't work until you update. You should also probably remove `changelogHeaderMessage` after updating since `changelogPreset.header` will be pulled and `changelogHeaderMessage` will be discarded. Good luck with your project, cheers
